scripts/PHPUnit/phpunit.xml

```diff
@@ -1,8 +1,29 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/10.5/phpunit.xsd" bootstrap="../../vendor/composer/vendor/autoload.php" colors="true" beStrictAboutOutputDuringTests="true" beStrictAboutTestsThatDoNotTestAnything="true" failOnRisky="true" failOnWarning="true" backupGlobals="true" executionOrder="random" cacheResultFile=".phpunit.cache/test-results">
-  <testsuites>
-    <testsuite name="all">
-        <directory>../../components/*/*/tests/</directory>
-    </testsuite>
-  </testsuites>
+<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/10.5/phpunit.xsd"
+         bootstrap="../../vendor/composer/vendor/autoload.php"
+         colors="true"
+         beStrictAboutOutputDuringTests="true"
+         beStrictAboutTestsThatDoNotTestAnything="true"
+         failOnRisky="true"
+         failOnWarning="true"
+         backupGlobals="true"
+         executionOrder="random"
+         cacheResultFile=".phpunit.cache/test-results"
+>
+    <testsuites>
+        <testsuite name="all">
+            <directory>../../components/*/*/tests/</directory>
+        </testsuite>
+    </testsuites>
+    <source>
+        <include>
+            <directory suffix=".php">../../components/*/*/</directory>
+        </include>
+        <exclude>
+            <directory>../../components/*/*/tests/</directory>
+            <directory>../../components/*/*/templates/</directory>
+            <directory>../../components/*/*/resources/</directory>
```

Comment:
Maybe we should add `libs` and `lib` here (e.g. for `components/ILIAS/Wiki/libs` and `components/ILIAS/UICore/lib`) as well as`artifacts`.